rust/gsc/src/main.rs

```diff
@@ -0,0 +1,51 @@
+use axum::{
+    // http::StatusCode,
+    routing::post,
+    Json,
+    Router,
+};
+use entity::job;
+use migration::{Migrator, MigratorTrait};
+use sea_orm::{ActiveModelTrait, ActiveValue, DatabaseConnection};
+use serde::Deserialize;
+use std::sync::Arc;
+
+#[derive(Deserialize)]
+struct AddJobPayload {
+    cmd: String,
+    env: String,
+}
+
+async fn add_job(Json(payload): Json<AddJobPayload>, conn: Arc<DatabaseConnection>) {
+    let insert_job = job::ActiveModel {
+        id: ActiveValue::NotSet,
+        cmd: ActiveValue::Set(payload.cmd),
+        env: ActiveValue::Set(payload.env),
+    };
+    insert_job.insert(conn.as_ref()).await.unwrap();
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // connect to our db
+
+    let connection = sea_orm::Database::connect("postgres://127.0.0.1/test").await?;
+    Migrator::up(&connection, None).await?;
+    let state = Arc::new(connection);
+
+    // build our application with a single route
+    let app = Router::new().route(
+        "/job",
+        post({
+            let shared_state = state.clone();
+            move |body| add_job(body, shared_state)
+        }),
+    );
+
+    // run it with hyper on localhost:3000
+    axum::Server::bind(&"127.0.0.1:3000".parse().unwrap())
```

Comment:
I realise this is early scaffold, but for later:

This relies on the host machine being able to route between the loopback device and other devices, and that the loopback hasn't been assigned to a WAN address. Can this be a less specific address like `0.0.0.0` or `::`, or to take the interface and/or ip address/port from a configuration file

Reply:
The plan is to take the ip address and port from an environment variable